sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/TracerProvider.java

```diff
@@ -172,10 +210,37 @@ public Context startSpan(String methodName, String databaseId, String endpoint,
                 }
             }).doOnSuccess(response -> {
                 if (isEnabled() && !isNestedCall) {
+                    ((Span) parentContext.get().getData(PARENT_SPAN_KEY).get()).makeCurrent();
+                    CosmosDiagnostics cosmosDiagnostics = diagnosticFunc.apply(response);
+                    try {
+                        Duration threshold = thresholdForDiagnosticsOnTracer;
+                        if(threshold == null) {
+                            threshold = CRUD_THRESHOLD_FOR_DIAGNOSTICS;
+                        }
+
+                        if (cosmosDiagnostics != null
+                            && cosmosDiagnostics.getDuration() != null
+                            && cosmosDiagnostics.getDuration().compareTo(threshold) > 0) {
+                            addDiagnosticsOnTracerEvent(cosmosDiagnostics);
+                        }
+                    } catch (JsonProcessingException ex) {
+                        LOGGER.debug("Error while serializing diagnostics for tracer", ex);
+                    }
                     this.endSpan(parentContext.get(), Signal.complete(), statusCodeFunc.apply(response));
                 }
             }).doOnError(throwable -> {
                 if (isEnabled() && !isNestedCall) {
+                    ((Span) parentContext.get().getData(PARENT_SPAN_KEY).get()).makeCurrent();
+                    Throwable unwrappedException = reactor.core.Exceptions.unwrap(throwable);
+                    if (unwrappedException instanceof CosmosException) {
+                        CosmosException dce = (CosmosException) unwrappedException;
+                        try {
+                            addDiagnosticsOnTracerEvent(dce.getDiagnostics());
+                        } catch (JsonProcessingException ex) {
+                            LOGGER.debug("Error while serializing diagnostics for tracer", ex);
```

Comment:
this is a failure case, any reason this is not warning? same for other places

Reply:
I was debating myself, i.e. if for any reason , cx start getting this on every request it will fill the logs. However we are not expecting this at all , so lets have warning . Done